service/kas/access/rewrap.go

```diff
@@ -134,29 +136,161 @@ func generateHMACDigest(ctx context.Context, msg, key []byte, logger logger.Logg
 	return mac.Sum(nil), nil
 }
 
-var acceptableSkew = 30 * time.Second
+func jwkFingerprintAttr(key jwk.Key) slog.Attr {
+	if key == nil {
+		return slog.String("jwk_thumbprint", "none")
+	}
+	thumbprint, err := key.Thumbprint(crypto.SHA256)
+	if err != nil {
+		return slog.String("jwk_thumbprint_error", err.Error())
+	}
+	return slog.String("jwk_thumbprint", base64.RawURLEncoding.EncodeToString(thumbprint))
+}
 
-func verifySRT(ctx context.Context, srt string, dpopJWK jwk.Key, logger logger.Logger) (string, error) {
-	token, err := jwt.Parse([]byte(srt), jwt.WithKey(jwa.RS256, dpopJWK), jwt.WithAcceptableSkew(acceptableSkew))
+// logSRTParseFailure emits contextual details when an SRT cannot be parsed, adjusting
+// verbosity depending on whether signature verification was required.
+func (p *Provider) logSRTParseFailure(ctx context.Context, srt string, requireVerification bool, err error) {
+	if p.Logger == nil {
+		return
+	}
+
+	attrs := []any{slog.Any("error", err)}
+	if requireVerification {
+		attrs = append(attrs, slog.String("srt", srt))
+		p.Logger.WarnContext(ctx, "unable to verify request token", attrs...)
+		return
+	}
+
+	p.Logger.WarnContext(ctx, "unable to validate or parse token", attrs...)
```

Comment:
```suggestion
	if requireVerification {
		attrs = append(attrs, )
		p.Logger.WarnContext(ctx, "unable to verify request token",
				slog.Any("error", err),
				slog.String("srt", srt))
		return
	}

	p.Logger.WarnContext(ctx, "unable to validate or parse token", slog.Any("error", err))
```

Reply:
@dmihalcik-virtru The log statement above this is wrapped in a conditional. What are you proposing here?